lms/djangoapps/courseware/views.py

```diff
@@ -68,6 +69,9 @@
 import survey.views
 
 from util.views import ensure_valid_course_key
+from certificates.models import CertificateStatuses as cert_status, certificate_status_for_student
+from certificates.queue import XQueueCertInterface
```

Comment:
I'm concerned about importing these objects directly from the `certificates` app, because doing so requires the `courseware` app to rely too much on the implementation of `certificates`.

I would much prefer this to be encapsulated in a Python API within `certificates`.  Following conventions we've used elsewhere, you could put this in a module called `certificates/api.py`.  The two operations this API needs to support are: (1) retrieving the certificate status for a particular course/student combination, and (2) adding a certificate generation task to the queue.